python/dgl/graphbolt/utils/internal.py

```diff
@@ -35,13 +36,31 @@ def save_data(data, path, fmt):
         raise RuntimeError(f"Unsupported format: {fmt}")
 
     # Perform necessary conversion.
-    if fmt == "numpy" and isinstance(element, torch.Tensor):
-        element = element.cpu().numpy()
-    elif fmt == "torch" and isinstance(element, np.ndarray):
-        element = torch.from_numpy(element).cpu()
+    if fmt == "numpy" and isinstance(data, torch.Tensor):
+        data = data.cpu().numpy()
+    elif fmt == "torch" and isinstance(data, np.ndarray):
+        data = torch.from_numpy(data).cpu()
 
     # Save the data.
     if fmt == "numpy":
         np.save(path, data)
     elif fmt == "torch":
         torch.save(data, path)
+
+
+def get_npy_dim(npy_path):
+    """Get the dim of numpy file."""
+    with open(npy_path, "rb") as f:
+        magic_str = f.read(6)
```

Comment:
instead of checking the magic, I would do:

```
try:
  read_array_header_1_0(f)
except ValueError as error:
  try:
    read_array_header_2_0(f)
  except ValueError as error:
    raise ValueError(f"Invalid file format")
```

Such that you don't need to check magic numbers by just relying on the provided library, which lower the maintenance cost for our code.